### PR TITLE
perf: run uploader task in threads

### PR DESCRIPTION
**Pull request by drewbo**
## Changes proposed in this pull request:
- try threading uploader for faster uploads

## security considerations
None

### CODE REVIEW

**Review by svenaas — DISMISSED** [review]
This looks good to me, but I think it would be wise for @apburnes to review as well to have one more opinion on thread safety.

File: src/publishing/s3publisher.py [hunk 1]
```diff
@@ -168,24 +169,30 @@ def publish_to_s3(directory, base_url, site_prefix, bucket, federalist_config,
...
-                    logger.warning(
-                        f'... unable to upload {file.filename} due '
-                        f'to invalid characters in file name.'
-                    )
-                else:
-                    raise
-
-    # Delete files not needed any more
+    else:
+        with ThreadPoolExecutor(max_workers=8) as executor:
```
Comment (apburnes) [inline src/publishing/s3publisher.py:191]: Did you play with the `max_workers` to test performance? Should we turn it into a parameter to facilitate performance testing?

**Review by apburnes — COMMENTED** [review]
Question about concurrency testing on the uploads.

File: src/publishing/s3publisher.py (hunk 1 above)
Comment (drewbo) [reply src/publishing/s3publisher.py:191]: I only tried this single value. In other contexts, I've seen that 25-50 gets max performance but happy to at least make this a constant for easier testing

**Review by apburnes — APPROVED** [review]

**Pull request merged**